src/amo/components/Search/index.js

```diff
@@ -60,49 +77,66 @@ export class SearchBase extends React.Component {
       LinkComponent,
       count,
       enableSearchSort,
+      errorHandler,
       filters,
-      hasSearchParams,
       loading,
-      page,
+      paginationQueryParams,
       pathname,
       results,
     } = this.props;
-    const queryParams = this.props.queryParams ||
+
+    const page = parsePage(filters.page);
+
+    // If paginator params aren't specified, we convert filters instead.
+    const queryParams = paginationQueryParams ||
       convertFiltersToQueryParams(filters);
-    const paginator = count > 0 && hasSearchParams ? (
+
+    // TODO: Make the count need to be higher than one page's worth of
+    // add-ons.
+    const paginator = count > 0 ? (
       <Paginate
         LinkComponent={LinkComponent}
-        currentPage={page}
         count={count}
+        currentPage={page}
         pathname={pathname}
         queryParams={queryParams}
       />
-    ) : [];
-    const searchSort = enableSearchSort && hasSearchParams && results.length ? (
+    ) : null;
+    const searchSort = enableSearchSort && count > 0 ? (
       <SearchSort filters={filters} pathname={pathname} />
     ) : null;
 
     return (
       <div className="Search">
+        {errorHandler.hasError() ? errorHandler.renderError() : null}
+
         <SearchContextCard />
 
         {searchSort}
 
         <SearchResults
           count={count}
           filters={filters}
-          hasSearchParams={hasSearchParams}
           loading={loading}
           pathname={pathname}
           results={results}
         />
+
         {paginator}
       </div>
     );
   }
 }
 
+export function mapStateToProps(state) {
+  return {
+    count: state.search.count,
+    loading: state.search.loading,
+    results: state.search.results,
+  };
+}
+
 export default compose(
-  safeAsyncConnect([{ promise: loadSearchResultsIfNeeded }]),
+  withErrorHandling({ name: 'Search' }),
```

Comment:
Ah. Those functions are *super* similarly named, it might be better to make them a bit easier to tell apart.

I've filed: #2931